proposal/interleaved-declarations.md

```diff
@@ -0,0 +1,125 @@
+# Interleaved Declarations: Draft 1
+
+*([Issue](https://github.com/sass/sass/issues/3846))*
+
+## Table of Contents
+
+* [Background](#background)
+* [Summary](#summary)
+* [Definitions](#definitions)
+  * [Current Style Rule](#current-style-rule)
+* [Declarations](#declarations)
+  * [Semantics](#semantics)
+* [Unknown At-Rules](#unknown-at-rules)
+  * [Semantics](#semantics-1)
+* [Deprecation Process](#deprecation-process)
+
+## Background
+
+> This section is non-normative.
+
+Historically, Sass has hoisted declarations that appear after nested rules into
+the original style rule. Although this doesn't match the logical ordering of the
+source document, it *does* avoid duplicating the style rule (and any other
+nested rules) for each set of interleaved properties.
+
+Many years after this decision was made, the [CSS Nesting Module] was published
+by the W3C, adding native support for nesting rules within style rules using
+essentially the same syntax as Sass. Although initially its semantics for
+interleaved declarations were the same as Sass's, [a later update] (four days
+ago at time of writing) changed it to preserve the logical ordering of
+interleaved declarations.
```

Comment:
I suspect this is fairly final. One way to think about it it that it went from "the CSSWG hadn't talked about this in depth and the editor just defaulted to the Sass behavior because it's definitely feasible if not ideal" to "the CSSWG has talked about this and found a way around the difficulties and now it's resolved".